mixer/adapter/list/list.go

```diff
@@ -79,7 +80,28 @@ func (h *handler) HandleListEntry(_ context.Context, entry *listentry.Instance)
 		return adapter.CheckResult{}, err
 	}
 
-	found, err := l.checkList(entry.Value)
+	var value string
+	isString := true
+
+	switch entryVal := entry.Value.(type) {
+	case *v1beta1.Value:
+		if entryVal.GetStringValue() != "" {
+			value = entryVal.GetStringValue()
+		} else if entryVal.GetIpAddressValue() != nil {
+			value = string(entryVal.GetIpAddressValue().Value[:])
```

Comment:
I think we have a bug in Mixer here.

Values of type IP_ADDRESS should be delivered to the adapter as an net.IP value and not as a byte[]. Should we fix this instead of continuing to do the wrong thing? Same holds for time and duration values, those should be delivered to adapters as time.Time and time.Duration. I don't know how they are delivered at the moment.

Reply:
Dynamic values are probably represented as raw attribute values (hence, the leakage of attribute model into adapter land). We will need to fix handling of dynamic wrappers across the code to hide raw `[]byte`.

Reply:
I guess the strong typing got lost when we switch to a lot of generated code. IP addresses definitely used to come in as net.IP as some point in the past